client/src/menu/main/PlayMenu.tsx

```diff
@@ -1,147 +1,188 @@
+                    if(e.key === 'Enter') {
+                        joinGame(roomCode, playerId);
+                    }
+                }}
+            />
         </div>
+        <button onClick={()=>{
+            joinGame(roomCode, playerId)
+        }}>
+            {translate("menu.play.button.join")}
+        </button>
+    </footer>
+}
+
+async function joinGame(roomCode?: number, playerId?: number): Promise<boolean> {
+    if (roomCode === undefined) return false;
+
+    Anchor.setContent(<LoadingScreen type="join"/>);
+
+    let success: boolean;
+    if (playerId === undefined) {
+        success = await GAME_MANAGER.sendJoinPacket(roomCode);
+    } else {
+        success = await GAME_MANAGER.sendRejoinPacket(roomCode, playerId);
+    }
+
+    if (success) {
+        Anchor.setContent(<LobbyMenu/>);
+    } else {
+        Anchor.setContent(<PlayMenu/>);
     }
+
+    return success;
+}
+
+async function hostGame() {
+    Anchor.setContent(<LoadingScreen type="host"/>);
+    if (await GAME_MANAGER.sendHostPacket()) {
+        Anchor.setContent(<LobbyMenu/>)
+    } else {
+        Anchor.setContent(<PlayMenu/>)
+    }
+}
+
+function refresh() {
+    GAME_MANAGER.sendLobbyListRequest();
+}
+
+type LobbyMap = Map<number, LobbyPreviewData>;
+
+function PlayMenuTable(): ReactElement {
+    const [lobbies, setLobbies] = useState<LobbyMap>(new Map());
+
+    useEffect(() => {
+        const listener: StateListener = (type) => {
+            if (GAME_MANAGER.state.stateType === "outsideLobby" && type === "lobbyList") {
+                setLobbies(GAME_MANAGER.state.lobbies);
+            }
+        }
+        GAME_MANAGER.addStateListener(listener);
+        return () => GAME_MANAGER.removeStateListener(listener);
+    });
+
+    return <table className="play-menu-table">
+        <thead>
+            <tr>
+                <th></th>
+                <th>{translate("menu.play.field.name")}</th>
+                <th>{translate("players")}</th>
+            </tr>
+        </thead>
+        <tbody>
+            {Object.entries(lobbies).map(entry=>{
+                const roomCode = Number.parseInt(entry[0]);
+                const lobby: LobbyPreviewData = entry[1];
+
+                return <tr key={roomCode}>
+                    <td><button onClick={() => joinGame(roomCode)}>{translate("menu.play.button.join")}</button></td>
+                    <td>{lobby.name}</td>
+                    <td>
+                        <div className="play-menu-lobby-player-list">
+                            {lobby.players.map((player, j)=>{
+                                return <button key={j} onClick={()=>{
+                                    joinGame(roomCode, player[0]);
+                                }}>{player[1]}</button>
+                            })}
+                        </div>
+                    </td>
+                </tr>;
+            })}
+        </tbody>
+        <tfoot>
+            {new Array(100).fill(0).map((_, i) => {
```

Comment:
100? I dont care enough to force you to fix this. But it should calculate the correct number since it might cause lag in the future.

Reply:
Ok, I won't do it